src/sx-design/src/Image/Image.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+
+import { warning } from '@adeira/js';
+import React, { useState, type Element } from 'react';
+import sx from '@adeira/sx';
+import { Blurhash } from 'react-blurhash';
+
+import useSxDesignContext from '../useSxDesignContext';
+
+type Props = {
+  +'width': number,
+  +'height': number,
+  +'src'?: string, // optional because BE might fail to return the URL
+  +'alt'?: Fbt,
+  +'blurhash'?: string,
+  +'onLoad'?: () => void,
+  +'onError'?: () => void,
+  +'data-testid'?: string,
+};
+
+export default function Image(props: Props): Element<'div'> {
+  const [isImageLoaded, setImageLoaded] = useState(false);
+  const { theme } = useSxDesignContext();
+
+  if (props.src != null) {
+    warning(
+      props.alt != null,
+      'You should specify alternative image text via `alt` property. This is an important ' +
+        'part of accessibility for screen reader users in order for them to understand the ' +
+        "content's purpose on the page.",
+    );
+  }
+
+  return (
+    <div
+      className={styles({
+        background: true,
+        backgroundLight: theme === 'light',
+        backgroundDark: theme !== 'light',
+      })}
+      style={{ width: props.width, height: props.height }}
+    >
+      {isImageLoaded === false && props.blurhash != null ? (
+        <Blurhash hash={props.blurhash} width={props.width} height={props.height} />
```

Comment:
This fails badly if the `blurhash` is invalid. I'd recommend validation of error handling

Reply:
Good catch! Fixed and added a test case for it. Thanks!